class.jetpack-jitm.php

```diff
@@ -332,7 +332,15 @@ function get_messages( $message_path, $query ) {
 			) );
 
 			$normalized_site_url      = Jetpack::build_raw_urls( get_home_url() );
-			$envelope->url            = 'https://jetpack.com/redirect/?source=jitm-' . $envelope->id . '&site=' . $normalized_site_url . '&u=' . $user->ID;
+
+			if ( ! class_exists( 'Jetpack_Affiliate' ) ) {
+				require_once JETPACK__PLUGIN_DIR . 'class.jetpack-affiliate.php';
+			}
+			// Get affiliate code and append it to the URL as aff=abc123
```

Comment:
I wonder if this may make sense to break things down here? As we add more parameters to that original `https://jetpack.com/redirect/?` URL, maybe we should build an array of parameters earlier and then add them to the root URL all at once? What do you think?

Reply:
I agree that using something like `add_query_arg()` would read a bit better.

We should probably also escape the URL since we're injecting variables into it.

Reply:
Updated, thanks for the review.